### PR TITLE
fix(core): fix parallel tests not being parallel

### LINKED ISSUE #764: Tests are not run in parallel
A missing `!` made all tests run in sequence. When adding the `!`, a lots of tests are failing. 
I'll try to sum up every tests here

### PR DESCRIPTION
**Pull request by Sh4d1**
Tests were not run in parallel :sweat_smile: 

Fixes #764 

Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

### CODE REVIEW

**Comment by Sh4d1** [comment]
Maybe disable it by defaut :thinking:

**Comment by QuentinBrosse** [comment]
> Maybe disable it by default 🤔

Most of our tests should pass in parallel has we are using cassets.

File: internal/namespaces/instance/v1/custom_server_test.go [hunk 1]
```diff
@@ -263,7 +266,8 @@ func Test_ServerDelete(t *testing.T) {
 			core.TestCheckGolden(),
 			core.TestCheckExitCode(0),
 		),
-		AfterFunc: core.ExecAfterCmd(`scw instance delete volume volume-id={{ (index .Server.Volumes "1").ID }}`),
+		AfterFunc:       core.ExecAfterCmd(`scw instance delete volume volume-id={{ (index .Server.Volumes "1").ID }}`),
```
Comment (remyleone) [inline internal/namespaces/instance/v1/custom_server_test.go]: It should be `scw instance volume delete`

File: internal/namespaces/instance/v1/custom_server_test.go [hunk 2]
```diff
@@ -252,7 +254,8 @@ func Test_ServerDelete(t *testing.T) {
 			core.TestCheckGolden(),
 			core.TestCheckExitCode(0),
 		),
-		AfterFunc: core.ExecAfterCmd(`scw instance delete volume volume-id={{ (index .Server.Volumes "0").ID }}`),
+		AfterFunc:       core.ExecAfterCmd(`scw instance delete volume volume-id={{ (index .Server.Volumes "0").ID }}`),
```
Comment (remyleone) [inline internal/namespaces/instance/v1/custom_server_test.go]: It should be `scw instance volume delete`

File: internal/human/marshal.go [hunk 3]
```diff
@@ -45,14 +45,17 @@ func Marshal(data interface{}, opt *MarshalOpt) (string, error) {
 
 	rType := rValue.Type()
 
+	// safely get the marshalerFunc
+	marshalerFunc, _ := getMarshalerFunc(rType)
```
Comment (loicbourgois) [inline internal/human/marshal.go]: why do we work on `marshalerFunc` ? 

is this necessary to fix the problem of parallel tests ? 

shouldn't this be in another PR ?

Comment (Sh4d1) [reply internal/human/marshal.go]: Changing the map to a sync map means changing the way we access the map, thus the getter :)

File: internal/namespaces/instance/v1/custom_server_create_test.go [hunk 4]
```diff
@@ -230,6 +230,7 @@ func Test_CreateServer(t *testing.T) {
 }
 
 // None of the tests below should succeed to create an instance.
+// these tests need to be run in sequence
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/custom_server_create_test.go]:
```suggestion
// these tests need to be run in sequence
```
```suggestion
// These tests need to be run in sequence.
```

File: internal/namespaces/instance/v1/custom_server_test.go [hunk 5]
```diff
@@ -223,14 +223,15 @@ func Test_ServerUpdateCustom(t *testing.T) {
...
-				core.ExecAfterCmd(`scw instance delete volume volume-id={{ (index .Server.Volumes "1").ID }}`),
+				core.ExecAfterCmd(`scw instance volume delete volume-id={{ (index .Server.Volumes "0").ID }}`),
+				core.ExecAfterCmd(`scw instance volume delete volume-id={{ (index .Server.Volumes "1").ID }}`),
 				deleteServer("Server"),
 			),
 		}))
 	})
 }
 
+// these tests needs to be run in sequence
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/custom_server_test.go]:
```suggestion
// These tests needs to be run in sequence.
```

File: internal/namespaces/instance/v1/custom_server_create_test.go (hunk 4 above)
Comment (loicbourgois) [inline internal/namespaces/instance/v1/custom_server_create_test.go]: Why ? 

Usually, comments should explain `why`, not `what`.

File: internal/namespaces/instance/v1/custom_server_test.go (hunk 5 above)
Comment (loicbourgois) [inline internal/namespaces/instance/v1/custom_server_test.go]: Why ? 

Comments should explain `why`, not `what`.

File: internal/namespaces/instance/v1/custom_server_create_test.go (hunk 4 above)
Comment (Sh4d1) [reply internal/namespaces/instance/v1/custom_server_create_test.go]: Good question :sweat_smile: it doesnt work in parallel but works in sequence. Not sure why :thinking:

File: internal/namespaces/instance/v1/custom_server_test.go (hunk 5 above)
Comment (Sh4d1) [reply internal/namespaces/instance/v1/custom_server_test.go]: Good question :sweat_smile: it doesnt work in parallel but works in sequence. Not sure why :thinking:

File: internal/namespaces/instance/v1/custom_server_create_test.go (hunk 4 above)
Comment (Sh4d1) [reply internal/namespaces/instance/v1/custom_server_create_test.go]: @loicbourgois fixed

**Pull request merged**